### PR TITLE
Use Slevomat Coding Standard method typehinting

### PR DESCRIPTION
**Pull request by xalopp**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
         "squizlabs/php_codesniffer": "^3.2.3",
         "escapestudios/symfony2-coding-standard": "^3.0",
         "dealerdirect/phpcodesniffer-composer-installer": "^0.4.3",
-        "php": ">=5.6.0"
+        "php": ">=5.6.0",
```
Comment (mmoll) [inline composer.json]: As Slevomat requires PHP 7.1, we could also bump this here.

**Comment by xalopp** [comment]
Using Slevomat means to drop PHP 5. Symfony 3.4 is is requires at least 5.5.9.

So this PR would be a breaking change.

Or shall we create a new branch to keep compat to 3.4 for 5.5.9?

**Comment by mmoll** [comment]
@xalopp Please see GH-84 for a separate PHP 7.1 change.

I'd suggest to merge GH-85 first and then branch off 2.x. But to be honest, personally I don't see any relationship between our CS and Symfony's PHP requirements. Even now we do require 5.6, which is >5.5.9.

Bringing in this new sniff here would require a major version bump anyway.

File: README.md [hunk 2]
```diff
@@ -35,7 +35,10 @@ The MO4 Coding Standard is an extension of the [Symfony Coding Standard](http://
 * There must be at least one space around operators, and (except for aligning multiline statements) at most one, see the
   [respective Squizlabs Sniff](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Customisable-Sniff-Properties#squizwhitespaceoperatorspacing)
   we imported with `ignoreNewlines = false`
-* Single quotes must be used instead of double quoutes, where possible.
+* Single quotes must be used instead of double quotes, where possible.
+* all method requires a return type; use `void` if the method returns nothing.
```
Comment (mmoll) [inline README.md]: "all methods require"

File: README.md [hunk 3]
```diff
@@ -35,7 +35,10 @@ The MO4 Coding Standard is an extension of the [Symfony Coding Standard](http://
...
+* all method parameters that defaults to null must be marked as nullable with the `?` symbol.
```
Comment (mmoll) [inline README.md]: "that default" and maybe use backticks for the `null`

File: README.md (hunk 2 above)
Comment (Ma27) [reply README.md]: And for consistency reasons with the bullet points above I'd vote for uppercase first :)

**Comment by mmoll** [comment]
I ran this aginst a small modern codebase and the results are pretty great :+1:

**Comment by mmoll** [comment]
@xalopp could you please also add `SlevomatCodingStandard.TypeHints.ReturnTypeHintSpacing`?

**Pull request merged**